macosx/Controller.mm

```diff
@@ -359,6 +359,61 @@ + (void)initialize
     }
 }
 
+void onStartQueue(tr_session* session, tr_torrent* tor, void* vself)
+{
+    auto* controller = (__bridge Controller*)(vself);
+    auto const hashstr = @(tr_torrentView(tor).hash_string);
+    auto* torrent = [controller torrentForHash:hashstr];
```

Comment:
Here and in other new functions below: the `torrentForHash` method accesses the `fTorrents` property, which is only safe to do from the main thread. I'm not sure where this particular line is executed on, but the dispatch below suggests it might not necessarily be the main thread at all times. I suggest we dispatch this line to the main queue as well just to be safe

Reply:
Sounds good. Possibly resolved by cd4357e, though I'm not familiar with these mechanics and would appreciate a re-review of this section :bow: 

It's not clear to me how `hashstr` is passed safely from one thread to the other, nor whether we need to manually do anything to clean it up when we're done with it -- e.g. will it be freed automatically, or is it leaked?

Reply:
We don't have to do anything - objc runtime takes care of `retain`/`release` calls by itself ever since I moved the project to use Automated Reference Counting. Basically you can think of any objc object reference (including `NSString *`) as of `shared_ptr` which is destroyed when it's out of scope as long as there are no other strong references to it.